### DISCUSSION TITLE
Guideline: Overloading methods.

### DISCUSSION THREAD

**Discussion by waldo1001** [microsoft/alguidelines#62] [BC Best Practices]
```AL
tableextension 53100 "Customer Ext BASE" extends Customer
{
    procedure BlockCustomer(HideDialog: Boolean)
    var
        WLDBlockCustomerMeth: Codeunit "WLD BlockCustomer Meth";
    begin
        WLDBlockCustomerMeth.BlockCustomer(Rec, HideDialog);
    end;

    procedure BlockCustomer()
    begin
        BlockCustomer(false);
    end;
}
```

**Comment by epernst** [comment]
Yes, let's get some guidelines on this one!
 When is it ok to use overlays and when NOT?
I have seen code with more than 10 overlays to the same function, in which case overlays does not make the code neither cleaner or easier to read.

**Comment by TheDoubleH** [comment]
@epernst - Depending on the scenario, I would argue the opposite. In NAV I would use a VARIANT on multiple occasions, when my code was semi-generic (i.e. support for Sales Header, Sales Line, Customer or Contact as a parameter.) - I would argue, that in that case, having overlays, makes the code easier for another developer to know what tables are currently 'supported'.

Granted, in my case, it is always just one record being passed.
